test/performance/priority.cpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2020-2022 Joel E. Anderson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <stumpless.h>
+#include "test/helper/memory_counter.hpp"
+
+NEW_MEMORY_COUNTER( prival_from_string )
+
+static void PrivalFromString(benchmark::State& state) {
+  const char *string = "user.info";
+
+  INIT_MEMORY_COUNTER( prival_from_string );
+
+  // Measure the time it takes to call the function.
+  for (auto _ : state) {
+    int result = stumpless_prival_from_string(string);
```

Comment:
This loop currently benchmarks the function with the input of `user.info`.

At a minimum, there should also be a benchmark that measures the performance of the function when a string containing a number is provided to it. It may make more sense for this to be a separate benchmark in the same `priority.cpp` file.

Also consider how you could benchmark more than just one of the possible input strings. Building an array of possible inputs before starting the `for` loop is one way to do this. It doesn't necessarily need to be comprehensive, but it would be nice to exercise a few more possible inputs.

Reply:
Yes, I suggest to use this list of input strings: ["191", "user.emerg", "uucp.err", "local7.debug"].
Every string in this list has different run time so benchmark would more valuable.

Reply:
Thanks for your input. 

Updated benchmark scores 

![image](https://github.com/goatshriek/stumpless/assets/54947026/13137565-05e1-49ab-a991-12945d495328)